src/mantine-hooks/src/use-counter/use-counter.ts

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+import { clamp } from '../utils';
+
+const DEFAULT_OPTIONS = {
+  min: -Infinity,
+  max: Infinity,
+};
+
+export function useCounter(
+  initialValue: number = 0,
+  options?: Partial<{ min: number; max: number }>
+) {
+  const { min, max } = { ...DEFAULT_OPTIONS, ...options };
+  const [count, setCount] = useState<number>(clamp({ value: initialValue, min, max }));
+
+  const inc = (delta: number = 1) => {
+    setCount((prevCount) => {
+      const newValue = prevCount + delta;
+      return clamp({ value: newValue, min, max });
+    });
+  };
+
+  const dec = (delta: number = 1) => {
```

Comment:
useless type information, it will be inferred from default value, `const dec = (delta = 1) => {`